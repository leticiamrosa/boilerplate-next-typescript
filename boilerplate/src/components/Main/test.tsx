import { render, screen } from '@testing-library/react'

import Main from '.'

describe('Main component', () => {
  it('should render the heading', () => {
    // given

    // when
    const { container } = render(<Main />)

    // then
    expect(
      screen.getByRole('heading', { name: /react avan√ßado/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
