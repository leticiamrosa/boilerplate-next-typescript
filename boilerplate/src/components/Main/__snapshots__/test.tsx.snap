// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component should render the heading 1`] = `
<main
  class="styles__Wrapper-dzx1ao-0 dvtrBm"
>
  <img
    alt="Imagem de um átomo e React Avançado escrito ao lado"
    class="styles__Logo-dzx1ao-1 dcISKB"
    src="/img/logo.svg"
  />
  <h1
    class="styles__Title-dzx1ao-2"
  >
    React Avançado
  </h1>
  <h2
    class="styles__Description-dzx1ao-3 edoRUQ"
  >
    Typescript, ReactJS, NextJS e Styled Components
  </h2>
  <img
    alt="Um desenvolvedor de frente para uma tela com código"
    class="styles__Illustration-dzx1ao-4 cSlyKt"
    src="/img/hero-illustration.svg"
  />
</main>
`
